rskj-core/src/test/java/co/rsk/peg/bitcoin/UtxoUtilsTest.java

```diff
@@ -0,0 +1,213 @@
+        List<Arguments> arguments = new ArrayList<>();
+
+        // -100, -200, -300
+        final byte[] negativeOutpointValues = Hex.decode(
+            "FF9CFFFFFFFFFFFFFFFF38FFFFFFFFFFFFFFFFD4FEFFFFFFFFFFFF");
+        String expectedMessageForNegativeOutpointValues = String.format(
+            "Invalid outpoint value: %s. Negative and null values are not allowed.", -100);
+        arguments.add(
+            Arguments.of(negativeOutpointValues, expectedMessageForNegativeOutpointValues));
+
+        // 100, 200, 300, -400
+        final byte[] negativeAndPositiveOutpointValues = Hex.decode("64C8FD2C01FF70FEFFFFFFFFFFFF");
+        String expectedMessageForNegativeAndPositiveOutpointValues = String.format(
+            "Invalid outpoint value: %s. Negative and null values are not allowed.", -400);
+        arguments.add(Arguments.of(negativeAndPositiveOutpointValues,
+            expectedMessageForNegativeAndPositiveOutpointValues));
+
+        final byte[] invalidOutpointValues = Hex.decode("FC9145DC00FAFF00FE");
+        String expectedMessageForInvalidOutpointValues = String.format(
+            "Invalid value with invalid VarInt format: %s", "FC9145DC00FAFF00FE");
+        arguments.add(Arguments.of(invalidOutpointValues, expectedMessageForInvalidOutpointValues));
+
+        final byte[] anotherInvalidOutpointValues = Hex.decode("FB8267DC00FCFF00FE");
+        String expectedMessageForAnotherInvalidOutpointValue = String.format(
+            "Invalid value with invalid VarInt format: %s", "FB8267DC00FCFF00FE");
+        arguments.add(Arguments.of(anotherInvalidOutpointValues,
+            expectedMessageForAnotherInvalidOutpointValue));
+
+        return arguments.stream();
+    }
+
+    @ParameterizedTest
+    @MethodSource("validOutpointValues")
+    void decodeOutpointValues(byte[] encodedValues, List<Coin> expectedDecodedValues) {
+        // act
+        List<Coin> decodeOutpointValues = UtxoUtils.decodeOutpointValues(encodedValues);
+
+        // assert
+        assertArrayEquals(expectedDecodedValues.toArray(), decodeOutpointValues.toArray());
+    }
+
+    @Test
+    void decodeOutpointValues_nullEncodedValues_shouldReturnEmptyList() {
+        // act
+        List<Coin> decodeOutpointValues = UtxoUtils.decodeOutpointValues(null);
+
+        // assert
+        List<Coin> expectedDecodedValues = Collections.EMPTY_LIST;
+        assertArrayEquals(expectedDecodedValues.toArray(), decodeOutpointValues.toArray());
+    }
+
+    @ParameterizedTest
+    @MethodSource("validOutpointValues")
+    void encodeOutpointValues(byte[] expectedEncodedOutpointValues, List<Coin> outpointValues) {
+        // act
+        byte[] encodeOutpointValues = UtxoUtils.encodeOutpointValues(outpointValues);
+
+        // assert
+        assertArrayEquals(expectedEncodedOutpointValues, encodeOutpointValues);
+    }
+
+    @Test
+    void decodeOutpointValues_nullOutpointValues_shouldReturnEmptyArray() {
```

Comment:
this test does exactly the same as [this](https://github.com/rsksmart/rskj/pull/2328/files#diff-bc219d6855f7bc8d87e68b50b242d584b2b4b7268cd3b2ed97e230c1b22178c3R157) one.
I think in this one you were supposed to test `encodeOutpointValues`

Reply:
Fixed